model/task_lifecycle.go

```diff
@@ -581,74 +596,57 @@ func MarkEnd(t *task.Task, caller string, finishTime time.Time, detail *apimodel
 	const maxStackTraceSize = 5
 	dependentTasks, err := t.FindAllUnmarkedBlockedDependencies()
 	if err != nil {
-		return errors.Wrapf(err, "getting tasks depending on task '%s'", t.Id)
-	}
-
-	if t.IsPartOfSingleHostTaskGroup() && len(dependentTasks) > 0 {
-		depTaskIds := []string{}
-		for _, depTask := range dependentTasks {
-			depTaskIds = append(depTaskIds, depTask.Id)
-		}
-		stack := message.NewStack(2, "").Raw().(message.StackTrace).Frames
-		if len(stack) > maxStackTraceSize {
-			stack = stack[:maxStackTraceSize-1]
-		}
-		grip.Info(message.Fields{
-			"message":                       "blocked task group dependent tasks",
-			"ticket":                        "EVG-12923",
-			"task":                          t.Id,
-			"execution":                     t.Execution,
-			"num_dependent_tasks_to_update": len(dependentTasks),
-			"dependent_task_ids_to_update":  depTaskIds,
-			"stack":                         stack,
-		})
+		return nil, errors.Wrapf(err, "getting tasks depending on task '%s'", t.Id)
+	}
+
+	depTaskIds := []string{}
+	for _, depTask := range dependentTasks {
+		depTaskIds = append(depTaskIds, depTask.Id)
 	}
 
 	for _, dependentTask := range dependentTasks {
 		if err = dependentTask.MarkUnattainableDependency(t.Id, true); err != nil {
-			return errors.Wrap(err, "marking dependency unattainable")
+			return nil, errors.Wrap(err, "marking dependency unattainable")
 		}
-		if err = UpdateBlockedDependencies(&dependentTask); err != nil {
-			return errors.Wrapf(err, "updating blocked dependencies for '%s'", t.Id)
+		innerDepTaskIds, err := UpdateBlockedDependencies(&dependentTask)
+		if err != nil {
+			return nil, errors.Wrapf(err, "updating blocked dependencies for '%s'", t.Id)
 		}
+		depTaskIds = append(depTaskIds, innerDepTaskIds...)
 	}
-	return nil
+	return depTaskIds, nil
 }
 
-func UpdateUnblockedDependencies(t *task.Task, logIDs bool, caller string) error {
+// UpdateUnblockedDependencies recursively marks all unattainable dependencies as attainable, and
+// returns a list of modified tasks IDs.
+func UpdateUnblockedDependencies(t *task.Task) ([]string, error) {
 	blockedTasks, err := t.FindAllMarkedUnattainableDependencies()
 	if err != nil {
-		return errors.Wrap(err, "getting dependencies marked unattainable")
+		return nil, errors.Wrap(err, "getting dependencies marked unattainable")
 	}
 
-	if logIDs && t.IsPartOfSingleHostTaskGroup() && len(blockedTasks) > 0 {
-		blockedTaskIds := []string{}
-		for _, blockedTask := range blockedTasks {
-			blockedTaskIds = append(blockedTaskIds, blockedTask.Id)
-		}
-		grip.Debug(message.Fields{
-			"message":          "unblocked task group dependent tasks",
-			"ticket":           "EVG-12923",
-			"blocked_task_ids": blockedTaskIds,
-			"task":             t.Id,
-			"caller":           caller,
-		})
+	blockedTaskIds := []string{}
+	for _, blockedTask := range blockedTasks {
+		blockedTaskIds = append(blockedTaskIds, blockedTask.Id)
```

Comment:
Same question as in `UpdateBlockedDependencies` about adding the task IDs in the for loop below this vs. adding it in a separate loop.